src/rgw/rgw_acl.h

```diff
@@ -15,95 +16,165 @@
 
 #include "rgw_basic_types.h" //includes rgw_acl_types.h
 
-class ACLGrant
-{
-protected:
-  ACLGranteeType type;
+// acl grantee types
+struct ACLGranteeCanonicalUser {
   rgw_user id;
-  std::string email;
-  mutable rgw_user email_id;
-  ACLPermission permission;
   std::string name;
-  ACLGroupTypeEnum group;
+
+  friend auto operator<=>(const ACLGranteeCanonicalUser&,
```

Comment:
why do we need to explicitly declare the default spaceship operator?

Reply:
the same page says:
> Like defaulted special member functions, a defaulted comparison function is defined if odr-used or needed for constant evaluation.

but i guess that if it does not compile, we have little to do

Reply:
where it says 'defaulted', i assume that means declared with =default